types/datasource.d.ts

```diff
@@ -13,9 +13,36 @@ import {
 } from './model';
 import {EventEmitter} from 'events';
 import {IsolationLevel, Transaction} from './transaction-mixin';
+import { ColumnMetadata, ConnectorSettings, ModelBase, ModelSettings, PersistedModel } from '..';
+import { PersistedModelClass } from './persisted-model';
+
+export type OperationOptions = {
+  accepts: string[],
+  returns: string[],
+  http: object,
+  remoteEnabled: boolean,
+  scope: unknown,
+  fnName: string,
```

Comment:
Refine TS typedef according to https://github.com/loopbackio/loopback-connector-openapi/blob/88cbea3ebcfe0fd7634e3fb49245bf8c15aa6a7a/README.md#extend-a-model-to-wrapmediate-api-operations

Reply:
Refine `scope` according to https://github.com/loopbackio/loopback-datasource-juggler/blob/be09d5774ef9f03c2a4c9c1c422a48b17b3a89df/lib/datasource.js#L187.

```suggestion
  accepts: string[],
  returns: string[],
  http: object,
  remoteEnabled: boolean,
  scope: DataAccessObject,
  fnName: string,
```

Note that we'll need to write `DataAccessObject` itself, which is only used to be mixin-ed into `DataSource` and `BaseModel`.